### DISCUSSION TITLE
Possible to override buildkite manager default fileMatch?

### DISCUSSION THREAD

**Discussion by lyonlai** [renovatebot/renovate#23529] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate/35.159.7-slim

### If you're self-hosting Renovate, select which platform you are using.

github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Make `fileMatch` configuration specified via the manager attribute able to override/ignore the default `fileMatch` attribute for buildkite manager. With the current configuration, buildkite manager is picking up unexpected files within the `.buildkite/scripts` sub-folder. 

Also we will need to employ a different manager for the `.buildkite/scripts` folder later down the road. hence, so `ignorePath` cannot be used in this case. 

### What you tried so far.

specified manager config with intended configuration. However the `.buildkite/scripts/config.yaml` is still being picked up. 

the config: 

```
{
  buildkite: {
  "fileMatch": [
      "buildkite\\.ya?ml",
      "\\.buildkite/[^/]+\\.ya?ml$",  // this line is expected to exclude .buildkite/script/config.yaml 
    ],
  }
}
```

### Relevant debug logs

<details><summary>Logs</summary>

```
{"name":"renovate",...,"msg":"Using file match: buildkite\\.ya?ml for manager buildkite"}
{"name":"renovate",...,"msg":"Using file match: \\.buildkite/.+\\.ya?ml$ for manager buildkite"}
{"name":"renovate",...,"msg":"Using file match: buildkite\\.ya?ml for manager buildkite"}
{"name":"renovate",...,"msg":"Using file match: \\.buildkite/[^/]+\\.ya?ml$ for manager buildkite"}
...
{"name":"renovate",,"msg":"Matched 2 file(s) for manager buildkite: .buildkite/pipeline.yml, .buildkite/scripts/config.yaml"}

```

</details>

**Comment by rarkins** [comment, marked as answer]
`fileMatch` is currently _additive_ so you can't override it. This is intentional so that users don't need to redefine the entire fileMatch just to add something to it.

I think ignorePaths can be nested inside buildkite, e.g.

```
{
  "buildkite": {
    "ignorePaths": [...]
  }
}
```

**Reply by lyonlai** [reply]
that works. thanks @rarkins
